### PR TITLE
Peer Code Review #1

### PR DESCRIPTION
**Pull request by FjPierre**
Is my code ready for merging onto production branch?

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,79 @@
-import logo from './logo.svg';
-import './App.css';
+// Modules
+import { useEffect, useState } from "react";
```
Comment (benbourgon) [inline src/App.js]: May want to remove useEffect here as it's not used in the App.js component

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,79 @@
...
+      "Monday",
+      "Tuesday",
+      "Wednesday",
+      "Thursday",
+      "Friday",
+      "Satuday",
+      "Sunday",
+    ];
+  };
+
```
Comment (benbourgon) [inline src/App.js]: Perhaps add comments to explain the purpose of these state variables

File: src/App.js [hunk 3]
```diff
@@ -1,25 +1,79 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <h1>Welcome to the weather app</h1>
+      <h2>Get the right temperature and dress according to the weather:</h2>
+
+      {/* Search bar to grab user location */}
```
Comment (benbourgon) [inline src/App.js]: I'm a big fan of spacing out attributes on different lines, nice and readable!

File: src/App.js [hunk 4]
```diff
@@ -1,25 +1,79 @@
...
+      <form
+        action=""
+        onSubmit={(e) => {
+          e.preventDefault();
+          if (cityName) {
```
Comment (benbourgon) [inline src/App.js]: Great error handling!

File: src/App.js [hunk 5]
```diff
@@ -1,25 +1,79 @@
...
+            setShouldFetch(true);
+          }
+        }}
+      >
+        <input
+          type="text"
+          className="search-bar"
+          placeholder="Select location..."
+          value={cityName}
+          onChange={(event) => {
```
Comment (benbourgon) [inline src/App.js]: Might want to move this anonymous function to be a "handleSearchBarInput" function or something similar.

File: src/App.js [hunk 6]
```diff
@@ -1,25 +1,79 @@
...
+            setCityName(event.target.value);
+          }}
+        />
+        <div className="buttonContainer">
+          <button>Get the weather</button>
+          {shouldFetch ? <FetchedDataDisplay {...{ cityName }} /> : null}
```
Comment (benbourgon) [inline src/App.js]: Cool conditional rendering!

File: src/App.js [hunk 7]
```diff
@@ -1,25 +1,79 @@
...
+          if (cityName) {
+            setShouldFetch(true);
+          }
+        }}
+      >
+        <input
+          type="text"
+          className="search-bar"
+          placeholder="Select location..."
+          value={cityName}
```
Comment (benbourgon) [inline src/App.js]: Might want to add a state variable to hold the user's input separate from the cityName state, which could get its value transferred to cityName after submit.

File: src/components/FetchedDataDisplay.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import { useEffect } from "react";
+import { useWeatherData } from "./useWeatherData";
+
+export function FetchedDataDisplay({ cityName }) {
+  const weatherData = useWeatherData(cityName);
+
```
Comment (benbourgon) [inline src/components/FetchedDataDisplay.js:5]: may want to add a comment to explain that fullName is the city's name here

File: src/components/FetchedDataDisplay.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import { useEffect } from "react";
```
Comment (benbourgon) [inline src/components/FetchedDataDisplay.js]: Could remove useEffect here as it's not in use in FetchedDataDisplay.js

File: src/components/FetchedDataDisplay.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    <div className="temperatureContainer">
+      <p>
+        The temperature today in {fullName} is {temperature} ℃
+      </p>
+    </div>
+  );
+}
+
+function getWeatherDetails(weatherData) {
+  if (weatherData !== null) {
```
Comment (benbourgon) [inline src/components/FetchedDataDisplay.js:18]: Nice error handling!

File: src/components/useWeatherData.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+
+// API call
+const api = {
```
Comment (benbourgon) [inline src/components/useWeatherData.js:4]: Very cool to have your key and base url on hand in an object, awesome.

File: src/components/useWeatherData.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  key: "44082ee5fdfb89d4da8cad9697c38e7f",
+  url: "https://api.openweathermap.org/data/2.5/weather",
+};
+
+export function useWeatherData(cityName) {
+  const [weatherData, setWeatherData] = useState(null);
+
+  // fetch the weather data on mount and save to state
+  useMount(() => {
```
Comment (benbourgon) [inline src/components/useWeatherData.js:13]: Very cool to have a "init" function wrapping your useEffect, I might have to borrow this...

File: src/components/useWeatherData.js [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+      // 1. get the coordinates of the city
+      const url = `http://api.openweathermap.org/geo/1.0/direct?q=${cityName}&appid=${api.key}`;
+
+      const resp = await fetch(url);
+      const data = await resp.json();
+
+      const { lat, lon } = data[0] ?? { lat: null, lon: null };
+
+      if (!lon) {
+        console.warn("OH NO no coords found");
```
Comment (benbourgon) [inline src/components/useWeatherData.js:24]: Accessing console warnings, first time I've heard of this!

**Review by benbourgon — APPROVED** [review]
This was awesome to take a look through, looks like a dry, clean, well put together app!

File: src/App.js (hunk 5 above)
Comment (FjPierre) [reply src/App.js]: Definitely a good idea and more practical. Thanks Ben

**Pull request merged**